test/integration_test.go

```diff
@@ -2516,6 +2518,191 @@ func (ts *IntegrationTestSuite) TestHeartbeatThrottleDisabled() {
 	ts.assertReportedOperationCount("temporal_request_failure", "RecordActivityTaskHeartbeat", 0)
 }
 
+func (ts *IntegrationTestSuite) TestUpsertMemoFromNil() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	systemInfo, err := ts.client.WorkflowService().GetSystemInfo(
+		ctx,
+		&workflowservice.GetSystemInfoRequest{},
+	)
+	ts.NoError(err)
+	if !systemInfo.GetCapabilities().GetUpsertMemo() {
+		ts.T().Skip("UpsertMemo not implemented in server yet")
+	}
+
+	upsertMemo := map[string]interface{}{
+		"key_1": "new_value_1",
+		"key_2": nil,
+		"key_3": 123,
+	}
+
+	expectedKey1Value, _ := converter.GetDefaultDataConverter().ToPayload("new_value_1")
+	expectedKey3Value, _ := converter.GetDefaultDataConverter().ToPayload(123)
+	expectedMemo := &commonpb.Memo{
+		Fields: map[string]*commonpb.Payload{
+			"key_1": expectedKey1Value,
+			"key_3": expectedKey3Value,
+		},
+	}
+
+	// Start workflow
+	wfid := "test-upsert-memo-from-nil"
+	wfOptions := ts.startWorkflowOptions(wfid)
+	run, err := ts.client.ExecuteWorkflow(ctx, wfOptions, ts.workflows.UpsertMemo, upsertMemo)
+	ts.NoError(err)
+	ts.NotNil(run)
+
+	var memo *commonpb.Memo
+	err = run.Get(ctx, &memo)
+	ts.NoError(err)
+
+	// Wait a little bit for ES to update
+	time.Sleep(2 * time.Second)
+
+	// Query ES for memo
+	var nextPageToken []byte
+	resp, err := ts.client.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
```

Comment:
I think a simple `DescribeWorkflowExecution` is much better here than listing. But I'm gonna mark this approved anyways, change only if you want.